### PR TITLE
XIVY-11675 Reinvent CodeEditor widget

### PR DESCRIPTION
**Pull request by ivy-lli**
New widgets
- ScriptInput (single line code editor, active on focus)
- ScriptArea (resizable code editor, always active)
- MacroInput (single line code editor, active on focus)
- MacroArea (Resizable code editor, active on focus)
- ScriptCell (with popover single line code editor)

Other stuff done:
- Link everywhere the label with the input (via useFieldset hook)
- Change test code for new CodeEditor widgets

### CODE REVIEW

**Comment by weissreto** [comment]
Cool implementations. 

As an inscription mask programmer, I may not be interested how a script editor is implemented (resizable) or how it behaves (focus). All I know is I need to have a macro input field, script input field, macro text area, script text area. How this is implemented is not important for me and may event change in future. By naming the basic buidling blocks independent of the implementation we can change the implementation later without the need to change all inscription mask once again.

Therefore I would suggest that we introduce these five widgets:
ScriptInput
ScriptArea (ScriptEditor?)
MacroInput
MacroArea (MacroEditor?)
ScriptCellEditor

These are used in the inscription masks. How they are implemented is hidden from the user and may change in future.

File: integrations/standalone/tests/integration/parts/call.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export class CallTester implements PartTest {
   }
   async fill(page: Page) {
     await ComboboxUtil.select(page, this.selectValue, this.selectLabel);
-    await TableUtil.fillExpression(page, 2, '"test"');
```
Comment (weissreto) [inline integrations/standalone/tests/integration/parts/call.ts:20]: I general we talk about the expression and not the code in mapping tables. 
I find the old naming better.

File: packages/editor/src/components/widgets/code-editor/index.ts [hunk 2]
```diff
@@ -1 +1,4 @@
 export { default as CodeEditor } from './CodeEditor';
+export { default as ResizableCodeEditor } from './ResizableCodeEditor';
+export { default as SingleLineCodeEditor } from './SingleLineCodeEditor';
```
Comment (ivy-rew) [inline packages/editor/src/components/widgets/code-editor/index.ts:3]: We can keep this name and it's very intuitive, but still a small remark.
In the process.model apis and bpm.exec we tried to distinguish single line codes from long statements by calling single liners "expression" (ScriptExpression, MacroExpression) and so on. 
If you like the term, you may adapt it here to, and name it "CodeExpressionEditor"?

File: integrations/standalone/tests/utils/table-util.ts [hunk 3]
```diff
@@ -19,10 +19,16 @@ export namespace TableUtil {
     }
   }
 
-  export async function fillExpression(page: Page, rowIndex: number, value: string) {
-    await page.locator('tbody').getByRole('row').nth(rowIndex).getByRole('textbox').click();
+  export async function fillCode(page: Page, rowIndex: number, value: string) {
```
Comment (weissreto) [inline integrations/standalone/tests/utils/table-util.ts:22]: either code or expression but not both

**Review by ivy-rew — APPROVED** [review]
Feels good, now we have concrete widgets almost as we have them in SWT ;-) ... so it feels like going into the same full featured direction :+1:

File: packages/editor/src/components/widgets/tag/Tags.css [hunk 4]
```diff
@@ -36,100 +36,3 @@
 .tag-add:focus-visible {
   box-shadow: var(--focus-shadow);
 }
-
```
Comment (weissreto) [inline packages/editor/src/components/widgets/tag/Tags.css:39]: Not needed anymore. some dead bodies here?

**Comment by ivy-rew** [comment]
> Cool implementations.
> 
> As an inscription mask programmer, I may not be interested how a script editor is implemented (resizable) or how it behaves (focus). All I know is I need to have a macro input field, script input field, macro text area, script text area. How this is implemented is not important for me and may even change in the future. By naming the basic building blocks independent of the implementation we can change the implementation later without the need to change all inscription mask once again.
> 
> Therefore I would suggest that we introduce these five widgets: ScriptInput ScriptArea (ScriptEditor?) MacroInput MacroArea (MacroEditor?) ScriptCellEditor
> 
> These are used in the inscription masks. How they are implemented is hidden from the user and may change in future.

I can second that the naming seems to technical right now. I like your proposals ... I'd go with the 'Input/area/cellEdit' naming. At least I'd instantly know which one too choose.

**Review by weissreto — CHANGES_REQUESTED** [review]

File: packages/editor/src/components/widgets/tag/Tags.css (hunk 4 above)
Comment (ivy-lli) [reply packages/editor/src/components/widgets/tag/Tags.css:39]: No, moved to Popover.css

File: integrations/standalone/tests/integration/parts/call.ts (hunk 1 above)
Comment (ivy-lli) [reply integrations/standalone/tests/integration/parts/call.ts:20]: ok i change it back 👍

File: integrations/standalone/tests/utils/table-util.ts (hunk 3 above)
Comment (ivy-lli) [reply integrations/standalone/tests/utils/table-util.ts]: same as above

**Review by ivy-rew — APPROVED** [review]

**Review by weissreto — APPROVED** [review]
Cool, thanks for the work

**Pull request merged**